src/rand.rs

```diff
@@ -432,3 +435,34 @@ mod fuchsia {
         fn zx_cprng_draw(buffer: *mut u8, length: usize);
     }
 }
+
+#[cfg(any(target_env = "sgx"))]
+mod sgx {
+    use core::arch::x86_64::_rdrand64_step;
+    use crate::error;
+
+    #[inline]
+    fn rdrand() -> Result<u64, error::Unspecified> {
```

Comment:
If you end up having multiple call-sites for this function (i.e. if you make the `chunks_exact_mut` changes), it might be worth having this function return a `[u8; 8]` instead of a `u64`.